test/test_torch.py

```diff
@@ -197,6 +197,16 @@ def test_sinh(self):
     def test_lgamma(self):
         self._testMathByName('lgamma')
 
+    @unittest.skipIf(not TEST_SCIPY, "Scipy not found")
+    def test_digamma(self):
+        from scipy.special import digamma
+        self._testMath(torch.digamma, digamma)
+
+    @unittest.skipIf(not TEST_SCIPY, "Scipy not found")
+    def test_trigamma(self):
+        from scipy.special import polygamma
```

Comment:
I've also started to wonder if we really want to expose trigamma instead of a more general `polygamma`. Can you change it to be consistent with SciPy?

Reply:
I like the idea of implementing `polygamma()` and I've looked into it, however it's much more complicated than `trigamma()`. Would you prefer a `polygamma(n,x)` that simply errors at `n>=2`?

Reply:
Note that Pymc3 has a `trigamma()` ([code](https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/special.py#L7)). If you do prefer `polygamma()` could you give me some tips on out how to change this PR? It was easy for me to implement `trigamma(x)` since I could copy wiring from @zou3519's `digamma(x)`, but `polygamma(n,x)` is a more complex interface and I'm not sure whether it will fit into the generic C macros and python tests.

Reply:
Yes I've been thinking of having a polygamma that only supports `n` up to 2 (or up to `1`, if we want to be consistent with SciPy). It should still be quite straightforward - just add an extra argument in all yaml declarations (`- int degree`), and add it to the C signature

Reply:
Ok thanks, I'll give that a try.

Reply:
Oh, degree might be a bad idea if we go with SciPy convention

Reply:
Ok I've updated this PR to expose `polygamma(n,x)` instead of `trigamma(x)`. CPU stuff works; I'm still working on CUDA stuff.